pkg/controller/testutil/test_utils.go

```diff
@@ -417,6 +419,19 @@ func (f *FakeRecorder) AnnotatedEventf(obj runtime.Object, annotations map[strin
 	f.Eventf(obj, eventtype, reason, messageFmt, args...)
 }
 
+func (f *FakeRecorder) AssertEventReasons(events []v1.Event) bool {
+	f.Lock()
+	defer f.Unlock()
+	equalEventReason := func(a *v1.Event, b v1.Event) bool {
+		if a.Reason != b.Reason {
+			f.T.Errorf("AssertList: expected %#v, got %#v", a.Reason, b.Reason)
```

Comment:
Why can't we use the existing pattern to output logs, like:
```
klog.FromContext(ctx).Error(err, "")
```
You've added `*testing.T` to the `FakeRecorder`, which has expanded the scope of changes (we could have kept it simple).

Reply:
Right! Reworked to not include testing.T